### ISSUE TITLE
AD0001 in SerializationRulesDiagnosticAnalyzer

### ISSUE THREAD

**Issue by KrisVandermotten** [dotnet/roslyn-analyzers#1454]
#### Analyzer package

Microsoft Code Analysis 2017 Version 2.3.0.62003

#### Analyzer

SerializationRulesDiagnosticAnalyzer

#### Actual behavior

```
Analyzer 'Microsoft.NetFramework.Analyzers.SerializationRulesDiagnosticAnalyzer' threw the following exception:
System.ArgumentException: Reported diagnostic with ID 'CA2235' is not supported by the analyzer.
Parameter name: diagnostic
   at Microsoft.CodeAnalysis.Diagnostics.DiagnosticAnalysisContextHelpers.VerifyArguments(Diagnostic diagnostic, Compilation compilationOpt, Func`2 isSupportedDiagnostic)
   at Microsoft.CodeAnalysis.Diagnostics.SymbolAnalysisContext.ReportDiagnostic(Diagnostic diagnostic)
   at Microsoft.NetFramework.Analyzers.SerializationRulesDiagnosticAnalyzer.SymbolAnalyzer.AnalyzeSymbol(SymbolAnalysisContext context)
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerExecutor.<>c.<ExecuteSymbolActionsCore>b__38_1(ValueTuple`2 data)
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerExecutor.ExecuteAndCatchIfThrows_NoLock[TArg](DiagnosticAnalyzer analyzer, Action`1 analyze, TArg argument, Nullable`1 info)
```

**Comment by jamesqo** [comment]
@KrisVandermotten Can you provide a minimal repro, please?

**Comment by mavasani** [comment]
I think I know the issue here. @KrisVandermotten were you using the Microsoft CodeAnalysis VSIX for analysis?

**Comment by mavasani** [comment]
Ah, I see that from your repro steps: Microsoft Code Analysis 2017 Version 2.3.0.62003

I will submit a fix soon..

**Comment by mavasani** [comment]
This should work once we release a new VSIX for VS2017.5. Keeping this open until the VSIX is published in the gallery.

**Comment by jinujoseph** [comment]
pending vsix release

**Comment by mavasani** [comment]
The extension for 15.5+ has been published: https://marketplace.visualstudio.com/items?itemName=VisualStudioPlatformTeam.MicrosoftCodeAnalysis2017

Note that next Visual Studio preview release should have resolved dotnet/roslyn#23766, at which point you should be able to have both the VSIX and NuGet package installed simultaneosly.
